daemon/containerd/progress.go

```diff
@@ -183,7 +189,13 @@ func (p *pushProgress) UpdateProgress(ctx context.Context, ongoing *jobs, out pr
 		id := stringid.TruncateID(j.Digest.Encoded())
 
 		status, err := p.Tracker.GetStatus(key)
-		if err != nil {
+
+		notStarted := (status.Total > 0 && status.Offset == 0)
+		if err != nil || notStarted {
+			if p.removeWaiting.Load() {
```

Comment:
A comment or more descriptive name here might be helpful, it took me a sec to figure out what this was doing

Reply:
Done